Amazon.QLDB.Driver/driver/QldbDriver.cs

```diff
@@ -292,42 +290,19 @@ internal T Execute<T>(Func<TransactionExecutor, T> func, RetryPolicy retryPolicy
                 }
                 catch (QldbTransactionException qte)
                 {
-                    this.driverBase.ThrowIfNoRetry(
+                    replaceDeadSession = this.driverBase.GetShouldReplaceDeadSessionOrThrowIfNoRetry(
                         qte,
                         session,
-                        retryPolicy.MaxRetries,
-                        ref replaceDeadSession,
-                        ref retryAttempt);
-
-                    try
-                    {
-                        retryAction?.Invoke(retryAttempt);
-                        Thread.Sleep(retryPolicy.BackoffStrategy.CalculateDelay(
-                            new RetryPolicyContext(retryAttempt, qte.InnerException)));
-                    }
-                    catch (Exception)
-                    {
-                        // Safeguard against semaphore leak if parameter actions throw exceptions.
-                        if (replaceDeadSession)
-                        {
-                            this.driverBase.PoolPermits.Release();
-                        }
-
-                        throw;
-                    }
+                        retryAttempt,
+                        retryPolicy,
+                        retryAction);
                 }
             }
         }
 
         internal QldbSession GetSession()
```

Comment:
I think GetSession() is only called in one place now, and as a one-liner I wonder if we still need it?

Reply:
It still serves a decent purpose and there's unit tests that depend on it.